src/boat/private/BoatConfigType.nim

```diff
@@ -0,0 +1,22 @@
+import std/parsecfg
+
+export parsecfg
+
+type BoatConfig* = ref object of RootObj
+  use*: string ## \
+    ## filepath, dir containing a file or remote uri
+
+  parsed: Config ## \
+    ## the parsed config after loading
+  parsedPath: string ## \
+    ## path on disk the parsed config was saved to
+
+proc `parsed=`*(self: BoatConfig, config: Config): void =
```

Comment:
this should accept a filepath and call parsecfg.loadPath filepath
we should be able to isolate parsecfg to this file